rdma: rename bounce to rx_buff

rauteric opened the pull request:
*Description of changes:*

Bounce buffers are used to receive both eager messages and ctrl messages -- in the latter case, the name "bounce buffer" doesn't make sense. Thus, rename to the more generic "rx_buff".

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

bwbarrett commented on include/nccl_ofi_param.h:
```diff
@@ -288,18 +288,18 @@ OFI_NCCL_PARAM_INT(disable_dmabuf, "DISABLE_DMABUF", 1);
+ * more rx buffers if we dip below this threshold, allocating new rx
  * buffers if needed.
  */
-OFI_NCCL_PARAM_INT(rdma_min_posted_bounce_buffers, "RDMA_MIN_POSTED_BOUNCE_BUFFERS", 64);
```
Let's leave this for right now (but add some comments about what it means).  You can think of this as an API break.  I think we want to have shadow naming (ie, have both options), but I don't want to block the rx buffer cleanup on that change.  We know we need to rewrite the parameter code, let's do this part with that change.

rauteric replied:
Done and comment added.

bwbarrett commented on include/nccl_ofi_rdma.h:
```diff
@@ -184,8 +184,8 @@ typedef struct nccl_net_ofi_rdma_close_msg {
-/* For LL/LL128 protocols, bounce buffers (source of RDMA read operations) need to be 128B aligned */
-#define BOUNCE_BUFFER_ALIGNMENT 128
+/* For LL/LL128 protocols, rx buffers (source of RDMA read operations) need to be 128B aligned */
+#define RX_BUFFER_ALIGNMENT 128
```
While we use it for both today, this really is all about bounce buffers.

rauteric replied:
Renamed to `EAGER_RX_BUFFER_ALIGNMENT`.

rauteric commented:
[Force-push](https://github.com/aws/aws-ofi-nccl/compare/3fdd9cd08252b100520f971890f9665d4e7d68ca..7cd6992bcaac7a3ebfc87c2e2f855a146089a9d8) to [7cd6992](https://github.com/aws/aws-ofi-nccl/commit/7cd6992bcaac7a3ebfc87c2e2f855a146089a9d8) addresses feedback.

[Force-push](https://github.com/aws/aws-ofi-nccl/compare/7cd6992bcaac7a3ebfc87c2e2f855a146089a9d8..40c765aeedf064a319803a2ae30ed6fc2c64cd13) to [40c765a](https://github.com/aws/aws-ofi-nccl/commit/40c765aeedf064a319803a2ae30ed6fc2c64cd13) is a rebase on master.
